### PR TITLE
Add a note about HTTP API call limits

### PR DESCRIPTION
**Pull request by burnash**

### CODE REVIEW

File: docs/user-guide.rst [hunk 1]
```diff
@@ -136,6 +136,30 @@ Get all values from the first column:
...
+
+    What's more, Sheets API v4 introduced `Usage Limits <https://developers.google.com/sheets/api/limits>`_
+    (500 requests per 100 seconds per project, and 100 requests per 100 seconds per user). When your
+    application hits that limit, you get an :meth:`~gspread.exceptions.APIError` `429 RESOURCE_EXHAUSTED`.
+
+    Here are the methods that may help you to reduce API calls:
+
+        * :meth:`~gspread.models.Worksheet.get_all_values` fetches values from all of the cells of the sheet.
+        * :meth:`~gspread.models.Worksheet.get` fetches all values from a range of cells.
+        * :meth:`~gspread.models.Worksheet.batch_get` can fetch values form multiple ranges of cells with one API call.
```
Comment (MichaelChirico) [inline docs/user-guide.rst:159]: form -> from

File: docs/user-guide.rst [hunk 2]
```diff
@@ -136,6 +136,30 @@ Get all values from the first column:
...
+    Under the hood, gspread uses `Google Sheets API v4 <https://developers.google.com/sheets/api>`_.
+    Most of the time when you call a gspread method to fetch or update a sheet gspread produces
+    one HTTP API call.
+
+    HTTP calls have performance costs. So if you find your app fetching values one by one in
+    a loop or iterating over rows or columns you can improve the performance of the app by fetching
+    data in one go.
+
+    What's more, Sheets API v4 introduced `Usage Limits <https://developers.google.com/sheets/api/limits>`_
+    (500 requests per 100 seconds per project, and 100 requests per 100 seconds per user). When your
```
Comment (MichaelChirico) [inline docs/user-guide.rst:152]: maybe "as of this writing"

**Comment by MichaelChirico** [comment]
LGTM! English is also fine :)

**Comment by burnash** [comment]
@MichaelChirico thank you very much for bringing this up and for the review.

**Comment by MichaelChirico** [comment]
happy to help in any small way 👍

**Pull request merged**